ComfortZoneApp/app/src/main/java/com/example/comfortzone/ui/HostActivity.java

```diff
@@ -96,37 +109,49 @@ private void createFragments() {
     private void listenerSetup() {
         bottomNavigationView.setOnItemSelectedListener(item -> {
             Pair<Integer, Integer> animations;
-            switch (item.getItemId()) {
-                case R.id.action_flight:
-                    fragment = flightFragment;
-                    animations = setAnimationLeftToRight();
-                    break;
-                case R.id.action_input:
-                    if (fragment == flightFragment) {
-                        animations = setAnimationRightToLeft();
-                    } else {
-                        animations = setAnimationLeftToRight();
-                    }
-                    fragment = inputFragment;
-                    break;
-                case R.id.action_profile:
-                default:
-                    fragment = profileFragment;
-                    animations = setAnimationRightToLeft();
-                    break;
+            if (isLoading) {
+                animations = setAnimationLeftToRight();
+                fragment = new LoadingFragment();
```

Comment:
you are rename this to something like currentFragment instead of the generic fragment name